### PR TITLE
fix: add function twice append provision bicep twice

### PR DESCRIPTION
**Pull request by wenytang-ms**
1. fix append configuration files.

E2E TEST: https://github.com/OfficeDev/TeamsFx/actions/runs/1533422769

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 1]
```diff
@@ -1641,13 +1641,11 @@ export class TeamsAppSolution implements Solution {
     const notifications: string[] = [];
     const pluginsToScaffold: LoadedPlugin[] = [this.LocalDebugPlugin];
     const azureResource = Array.from(settings.azureResources || []);
-    if (addFunc || ((addSQL || addApim) && !alreadyHaveFunction)) {
+    if (addFunc && !alreadyHaveFunction) {
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1644]: we may also need to handle cases on addFunc = true and alreadyHaveFunction = true, which only scaffolded an new api to Azure Functions project

Comment (wenytang-ms) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts]: sure, restore the logic

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 2]
```diff
@@ -1641,9 +1641,9 @@ export class TeamsAppSolution implements Solution {
     const notifications: string[] = [];
     const pluginsToScaffold: LoadedPlugin[] = [this.LocalDebugPlugin];
     const azureResource = Array.from(settings.azureResources || []);
-    if (addFunc || ((addSQL || addApim) && !alreadyHaveFunction)) {
-      pluginsToScaffold.push(functionPlugin);
+    if (addFunc || ((addSQL || addApim || addKeyVault) && !alreadyHaveFunction)) {
       if (!azureResource.includes(AzureResourceFunction.id)) {
```
Comment (adashen) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1645]: one question, it uses id to identify the scaffold. If the add is before provision, what will happen?

Comment (wenytang-ms) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1645]: these id read from azureResouruces in projectSettings.json file, provision has no effect on this value.

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts (hunk 2 above)
Comment (adashen) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1645]: Got it!

**Review by formulahendry — APPROVED** [review]

File: packages/fx-core/tests/plugins/solution/solution.arm.test.ts [hunk 3]
```diff
@@ -368,6 +368,10 @@ output teamsFxConfigurationOutput object = contains(reference(resourceId('Micros
     expect(
       await fs.readFile(path.join(projectArmTemplateFolder, "../provision/bot.bicep"), fileEncoding)
     ).equals("Mocked bot Provision content. simple auth endpoint: Mocked simple auth endpoint");
+
+    expect(
+      await fs.readFile(path.join(projectArmTemplateFolder, "../teamsFx/bot.bicep"), fileEncoding)
```
Comment (dilin-MS) [inline packages/fx-core/tests/plugins/solution/solution.arm.test.ts:373]: Can remove the above
```ts
expect(await fs.pathExists(path.join(projectArmTemplateFolder, "../teamsFx/bot.bicep"))).to.be.true;
```

Comment (wenytang-ms) [reply packages/fx-core/tests/plugins/solution/solution.arm.test.ts:373]: sure

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**